common/djangoapps/enrollment/views.py

```diff
@@ -26,7 +29,11 @@
     OAuth2AuthenticationAllowInactiveUser,
     SessionAuthenticationAllowInactiveUser
 )
-from openedx.core.lib.api.permissions import ApiKeyHeaderPermission, ApiKeyHeaderPermissionIsAuthenticated
+from openedx.core.lib.api.permissions import (
+    ApiKeyHeaderPermission,
+    ApiKeyHeaderPermissionIsAuthenticated,
+    IsUserInUrlOrStaff
```

Comment:
If we're only wanting to make a staff check, we should use (or create) that permission class specifically.  This permission class makes it possible to specify a user in the URL as a query parameter -- not sure we want to allow for that in this scenario.